11-algorithms.Rmd

```diff
@@ -82,19 +91,30 @@ See the web page reprex.tidyverse.org for details.
 \index{reproducibility}
 
 The contents of this section apply to any type of R script.
-A particular consideration with scripts for geocomputation is that they tend to have external dependencies, such as the QGIS\index{QGIS} dependency to run code in Chapter \@ref(gis), and require input data in a specific format.
+A particular consideration with scripts for geocomputation is that they tend to have external dependencies, such as the QGIS\index{QGIS} dependency to run code in Chapter \@ref(gis).
+Scripts for working with geographic data also often require input datasets to be available specific formats.
 Such dependencies should be mentioned as comments in the script or elsewhere in the project of which it is a part, as illustrated in the script [`11-centroid-alg.R`](https://github.com/geocompx/geocompr/blob/main/code/11-centroid-alg.R).
-The work undertaken by this script is demonstrated in the reproducible example below, which works on a pre-requisite object named `poly_mat`, a square with sides 9 units in length (the meaning of this will become apparent in the next section):^[
-This example shows that `source()` works with URLs (a shortened version is used here), assuming you have an internet connection.
-If you do not, the same script can be called with `source("code/11-centroid-alg.R")`, assuming you are running R from the root directory of the `geocompr` folder, which can be downloaded from https://github.com/geocompx/geocompr.
-]
+'Defensive' programming techniques and good error messages can save time by checking for dependencies and communicating with users if certain requirements are not met.
+If statements, implemented with `if()` in R, can be used to send messages or run lines of code if, and only if, certain conditions are met.
+The following lines of code, for example, send a message to users if a certain file is missing:
+
+
+```{r}
+if(!file.exists("required_geo_data.gpkg")) {
```

Comment:
not sure if we ever talked about it but I guess we are largely following the tidyverse style guide. And then we would write `if (!...)` instead of `if(!...)`.

Reply:
See https://style.tidyverse.org/syntax.html#parentheses

Reply:
See mega commit here: https://github.com/geocompx/geocompr/pull/923/commits/de5c085089135e40bda96b02dbf244c20ea5b03e